### ISSUE TITLE
cidrhost type detection broken in ternary (REGRESSION)

### ISSUE THREAD

**Issue by in4mer** [hashicorp/terraform#14597]
### Terraform Version
v0.9.5

### Affected Resource(s)
- core

### Configuration Files
This is throwing errors in a module used to create instances:
```
private_ip              = "${var.private_octet > "0" ? cidrhost(module.instance-subnet.cidr_block, var.private_octet) : "" }"
```

### Debug Output
Here's the trace gist https://gist.github.com/in4mer/dd68fa044f18de2e4ea590712389fa6e

### Panic Output
No panic

### Expected Behavior
The string should be computed per the ternary

### Actual Behavior
```
1 error(s) occurred:

* module.sftp.aws_instance.instance: 1 error(s) occurred:

* module.sftp.aws_instance.instance: At column 3, line 1: true and false expression types must match; have type unknown and type string in:

${var.private_octet > "0" ? cidrhost(module.instance-subnet.cidr_block, var.private_octet) : "" }
```

### Steps to Reproduce
I couldn't reproduce this error when using a test setup, even when calling aws_instance, but it happens upon every invocation of my instance module.

1. `terraform apply`

### Important Factoids
I'm calling aws_instance from within a module.  cidrhost should always return a string; wtf is up with this type detection?

I can't launch ANY instances, in stage or production, because I've been using this instance module since v0.8.  This was introduced in either v0.9.4 or v0.9.5.  IMO this is a regression.

### References
Couldn't find any references.

**Comment by apparentlymart** [comment]
Hi @in4mer! Sorry for this regression.

This looks like the same thing that was fixed in #14454. That will be included in the next release. Some workarounds for this, to keep moving before the next releases is out, were discussed in #14399.

Sorry again for this regression, and thanks for reporting it.

**Comment by in4mer** [comment]
TYVM!  +1
